qiskit/pulse/commands/acquire.py

```diff
@@ -113,52 +119,82 @@ class AcquireInstruction(Instruction):
 
     def __init__(self,
                  command: Acquire,
-                 acquires: Union[AcquireChannel, List[AcquireChannel]],
-                 mem_slots: Union[MemorySlot, List[MemorySlot]],
-                 reg_slots: Optional[Union[RegisterSlot, List[RegisterSlot]]] = None,
+                 acquire: Union[AcquireChannel, List[AcquireChannel]],
+                 mem_slot: Optional[MemorySlot] = None,
```

Comment:
Since mem_slots was a positional argument before, let's move `mem_slots` and `reg_slots` in front of the new arguments during the deprecation

Reply:
Ideally, both of these would work for now:
```
AcquireInstruction(cmd, AcquireChannel(0), MemorySlot(0))
AcquireInstruction(cmd, AcquireChannel(0), [MemorySlot(0)])
```
Maybe we have to keep `mem_slot`, remove `mem_slots`, remove ` = None`, and allow the type of `mem_slot` to be `Union[MemorySlot, List[MemorySlot]]`